DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs

```diff
@@ -432,6 +461,44 @@ private void LbMapList_SelectedIndexChanged(object sender, EventArgs e)
             ChangeMap(GameMode, map);
         }
 
+        private void PickRandomMap()
+        {
+            int totalPlayerCount = Players.Count(p => p.SideId < ddPlayerSides[0].Items.Count - 1)
+                   + AIPlayers.Count;
+            List<Map> maps = GetMapList(totalPlayerCount);
+            if (maps.Count < 1)
+                return;
+            Map newMap = maps[new Random().Next(0, maps.Count())];
+
+            tbMapSearch.Text = string.Empty;
+            tbMapSearch.OnSelectedChanged();
+            ListMaps();
+            SelectMapInList(newMap);
+        }
+
+        private List<Map> GetMapList(int playerCount)
+        {
+            List<Map> mapList = new List<Map>(GameMode.Maps.Where(x => x.MaxPlayers == playerCount));
+            if (mapList.Count < 1 && playerCount <= MAX_PLAYER_COUNT)
+                return GetMapList(playerCount + 1);
+            else
+                return mapList;
+        }
+
+        private void SelectMapInList(Map map)
+        {
+            for (int i = 0; i < lbMapList.ItemCount; i++)
+            {
+                Map listMap = (Map)lbMapList.GetItem(1, i).Tag;
+                if (listMap == map)
+                {
+                    lbMapList.SelectedIndex = i;
+                    while (i > lbMapList.LastIndex)
+                        lbMapList.TopIndex++;
```

Comment:
I updated the code so that ListMaps correctly sets the position on map list afterwards, which should fix the focus issue for both the search and changing game modes. The random map button also now relies on this rather than the now removed SelectMapInList method.